atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableTimestamps.java

```diff
@@ -78,39 +85,67 @@ private SweepableTimestampsTable.SweepableTimestampsColumn computeColumn(long st
      * sweepTs
      */
     Optional<Long> nextTimestampPartition(ShardAndStrategy shardStrategy, long lastSweptTs, long sweepTs) {
+        List<Long> partitions = nextTimestampPartitions(shardStrategy, lastSweptTs, sweepTs, 1);
+        if (partitions.isEmpty()) {
+            return Optional.empty();
+        } else {
+            return Optional.of(partitions.get(0));
+        }
+    }
+    // TODO: This is Jeremy's code - clean it up.
+    // Public because testing out visibility things
```

Comment:
We need to rethink this, I think the schema is actually a bit different from what we had thought here! Basically sweepableTimestamps indeed has one cell, but that cell can have a ton of versions

Reply:
actually ok if batch size is 1. but cannot use batch size > 1.